executor/union_scan.go

```diff
@@ -147,10 +152,7 @@ func (us *UnionScanExec) getSnapshotRow() (*Row, error) {
 			if us.snapshotRow == nil {
 				break
 			}
-			if len(us.snapshotRow.RowKeys) != 1 {
-				return nil, ErrRowKeyCount
-			}
-			snapshotHandle := us.snapshotRow.RowKeys[0].Handle
+			snapshotHandle := us.snapshotRow.Data[len(us.snapshotRow.Data)-1].GetInt64()
```

Comment:
us.handle may be nil if us's executor don't need to return the handle when call next.
Currently if there's a union scan, we will always put the handle into the tail although there may be a pk column.